osbuild/util/rhsm.py

```diff
@@ -5,18 +5,59 @@
 """
 
 import configparser
+import glob
+import os
 import re
 
 
 class Subscriptions:
     def __init__(self, repositories):
         self.repositories = repositories
+        self.fallback_method_available = False
+        self.fallback_secrets = None
+
+    @staticmethod
+    def _get_fallback_rhsm_secrets():
+        rhsm_secrets = {
+            'ssl_ca_cert': "/etc/rhsm/ca/redhat-uep.pem",
+            'ssl_client_key': "",
+            'ssl_client_cert': ""
+        }
+
+        keys = glob.glob("/etc/pki/entitlement/*-key.pem")
+        for key in keys:
+            # The key and cert have the same prefix
+            cert = key.rstrip("-key.pem") + ".pem"
+            # The key is only valid if it has a matching cert
+            if os.path.exists(cert):
+                rhsm_secrets['ssl_client_key'] = key
+                rhsm_secrets['ssl_client_cert'] = cert
+                return rhsm_secrets
+
+        raise RuntimeError("no matching rhsm key and cert")
 
     @classmethod
     def from_host_system(cls):
         """Read redhat.repo file and process the list of repositories in there."""
-        with open("/etc/yum.repos.d/redhat.repo", "r") as fp:
-            return cls.parse_repo_file(fp)
+        redhat_repo_not_found = False
+        fallback_secrets_not_found = False
+        ret = cls(None)
+        try:
+            with open("/etc/yum.repos.d/redhat.repo", "r") as fp:
+                ret = cls.parse_repo_file(fp)
+        except FileNotFoundError:
+            redhat_repo_not_found = True
```

Comment:
Not a big deal at all to keep it that way, but you could do:

```python3

with contextlib.suppress(FileNotFoundError):
    with open("/etc/yum.repos.d/redhat.repo", "r") as fp:
        ret = cls.parse_repo_file(fp)

with contextlib.suppress(RuntimeError):
    ret.fallback_secrets = cls._get_fallback_rhsm_secrets()

if not ret.repositories and not ret.fallback_secrets:
    raise RuntimeError("No RHSM secrets found on this host.")
```

One thing, since both of the methods are supported and only one required, they are both optional as long as one exists and thus maybe should not throw and exception since they are always ignored now anyway.

Reply:
Do you mean we should suppress all exceptions in those functions as long as the other one works?

Reply:
I mean, either use the code outlined above, or just get rid of the exceptions inside `cls.parse_repo_file` and `cls._get_fallback_rhsm_secrets` and then only throw the last one if we have neither `repos` nor `secrets`.

Reply:
ok, I've used the code outlined above, so this is done :-)